src/uu/date/src/date.rs

```diff
@@ -223,10 +223,17 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
                 Box::new(iter)
             }
             DateSource::File(ref path) => {
-                file = File::open(path).unwrap();
-                let lines = BufReader::new(file).lines();
-                let iter = lines.filter_map(Result::ok).map(parse_date);
-                Box::new(iter)
+                match File::open(path) {
+                    Ok(file) => {
+                        let lines = BufReader::new(file).lines();
+                        let iter = lines.filter_map(Result::ok).map(parse_date);
+                        Box::new(iter)
+                    }
+                    Err(_) => {
+                        eprintln!("Error: File not found at path {:?}", path);
+                        Box::new(std::iter::empty())
```

Comment:
could you please add a test to make sure we don't regress ?

Reply:
Sure! Where would the tests for this go?

Reply:
Sure, the tests should go in `test_date.rs` right? Can't see a test for file stuff on there.

Reply:
yeah